enterprise/frontend/src/metabase-enterprise/data-studio/segments/components/SegmentBreadcrumbs/SegmentBreadcrumbs.tsx

```diff
@@ -0,0 +1,144 @@
+import { skipToken } from "@reduxjs/toolkit/query";
+import { Fragment } from "react";
+import { Link } from "react-router";
+import { t } from "ttag";
+
+import {
+  useGetCollectionQuery,
+  useListDatabaseSchemasQuery,
+} from "metabase/api";
+import { Ellipsified } from "metabase/common/components/Ellipsified";
+import * as Urls from "metabase/lib/urls";
+import { getCollectionList } from "metabase/nav/components/CollectionBreadcrumbs/utils";
+import { Flex, Group, Text } from "metabase/ui";
+import type { Segment, Table } from "metabase-types/api";
+
+import S from "./SegmentBreadcrumbs.module.css";
+
+type SegmentBreadcrumbsProps = {
+  table: Table;
+  segment?: Segment;
+};
+
+export function PublishedTableSegmentBreadcrumbs({
+  table,
+  segment,
+}: SegmentBreadcrumbsProps) {
+  const { data: collection } = useGetCollectionQuery(
```

Comment:
@alxnddr table.collection should be hydrated in /api/table/:id and /api/table/:id/query_metadata already